docs/troubleshooting.md

```diff
@@ -0,0 +1,134 @@
+
+This case itself breaks down into three possible remedies, depending on whether you can move the
+require to your own code, the missing dependency can be considered a "bug",
+or the third-party package uses the "plugin pattern" to discover its
+plugins dynamically at runtime based on finding them based on a string you provided.
+
+#### The `require` can move to first-party
+
+This is the most principled solution. In many cases, a library that accepts the name of a package as
+a string will also accept it as an object, so you can refactor `config: ['some-package']` to
+`config: [require('some-package')]`. You may need to change from json or yaml config to a JavaScript
+config file to allow the `require` syntax.
+
+Once you've done this, it's handled like the "require appears in your code" case above.
+
+For example, the
+[documentation for the postcss-loader for Webpack](https://webpack.js.org/loaders/postcss-loader/#sugarss)
+suggests that you `npm install --save-dev sugarss`
+and then pass the string "sugarss" to the `options.postcssOptions.parser` property of the loader.
+However this violates symmetry and would require workarounds listed below.
+You can simply pass `require("sugarss")` instead of the bare string, then include the `sugarss`
+package in the `data` (runtime dependencies) of your `webpack.config.js`.
+
+#### It's a bug
+
+This is the case when a package has a `require` statement in its runtime code for some package, but
+it doesn't list that package in its `package.json`, or lists it only as a `devDependency`.
+
+pnpm and Yarn PnP will hit the same bug. Conveniently, there's already a shared database used by
+both projects to list these, along with the missing dependency edge:
+https://github.com/yarnpkg/berry/blob/master/packages/yarnpkg-extensions/sources/index.ts
+
+> We should use this database under Bazel as well. Follow
+> https://github.com/aspect-build/rules_js/issues/1215.
+
+The recommended fix for both pnpm and rules_js is to use
+[pnpm.packageExtensions](https://pnpm.io/package_json#pnpmpackageextensions)
+in your `package.json` to add the missing `dependencies` or `peerDependencies`.
+
+Example,
+
+https://github.com/aspect-build/rules_js/blob/a8c192eed0e553acb7000beee00c60d60a32ed82/package.json#L12
+
+> Make sure you pnpm install after changing `package.json`, as rules_js only reads the
+> `pnpm-lock.yaml` file to gather dependency information.
+> See [Fetch third-party packages](./README.md#fetch-third-party-packages-from-npm)
+
+#### It's a plugin
+
+Sometimes the package intentionally doesn't list dependencies, because it discovers them at runtime.
+`eslint` and `prettier` are common typical examples.
+
+The solution is based on pnpm's [public-hoist-pattern](https://pnpm.io/npmrc#public-hoist-pattern).
+Use the [`public_hoist_packages` attribute of `npm_translate_lock`](./npm_translate_lock.md#npm_translate_lock-public_hoist_packages). This makes the `require` statement appear in the
```

Comment:
"This makes the `require` statement appear in the "public" root of the `node_modules` tree" 

not quite right...

Reply:
removed that sentence